### PR TITLE
add step fail hook

### PR DESCRIPTION
**Pull request by HarrisChu**
https://discuss.nebula-graph.com.cn/t/topic/6707/2

#### What type of PR is this?
- [ ] bug
- [ ] feature
- [x] enhancement

#### What does this PR do?
currently, we just display error information in `cmp_dataset` function, if there's any error not in the function, we cannot find more details.
so add step error hook, and display space if using a random space.

image

### CODE REVIEW

**Review by SuperYoko — APPROVED** [review]
Nice

**Review by yixinglu — APPROVED** [review]

File: tests/conftest.py [hunk 1]
```diff
@@ -64,6 +65,16 @@ def pytest_addoption(parser):
                      help="Nebula Graph workspace")
 
 
+def pytest_bdd_step_error(request, feature, scenario, step, step_func, step_func_args):
```
Comment (darionyaphet) [inline tests/conftest.py:68]: `pytest_bdd_step_error` not by called?

Comment (HarrisChu) [reply tests/conftest.py:68]: it registered by pytest automatically.
1. pytest scan `conftest`
2. match the same definition with hook in pytest-bdd.

https://github.com/pytest-dev/pytest-bdd/blob/master/pytest_bdd/plugin.py

**Review by darionyaphet — APPROVED** [review]
LGTM

File: tests/conftest.py (hunk 1 above)
Comment (darionyaphet) [reply tests/conftest.py:68]: 👍

**Pull request merged**